### DISCUSSION TITLE
Enabling two-factor (2FA) in our on-premise set-up

### DISCUSSION THREAD

**Discussion by edward221096** [kimai/kimai#4220] [On-Premise: Installation & Updates]
Hi Guys,

I would like to seek assistance in setting up Kimai two-factor (2FA) in our on-premise setup. 

Below are our set-up:
1. We are using Laminas LDAP extension and configured it as Active directory as our main authentication provider
2. We do not have an existing 2FA set-up in place

Tried the following steps:
1. I logged in as a standard user
2. Go to 2FA > scanned the QR code in my iphone
3. Added a user, password, kimai website and activated the verification code in my iphone
4. Input the verification code from my iphone to our kimai website
5. Verification code worked > select activated > then save
6. Log out and logged in again but unfortunately there is no pop up for 2FA.

Question:
1. Do we have any available documentations or guide on how to set this up?
2. Is it possible to setup 2FA on-premise? (not using Kimai Cloud)
3. Do I missed something in setting it up based on the above steps?

Thanks,
Edward

**Comment by kevinpapst** [comment]
In theory it should work, but I never tested 2FA with LDAP, so I can't say whether it works or not.
